src/Orleans.Transactions/Abstractions/ITransactionalFaultInjector.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Orleans.Transactions
+{
+    public interface ITransactionalFaultInjector
```

Comment:
it is a bit odd to include the testing interface in TransactionAbstraction package. I run into the same problem in my PR #4874 . I'm thinking moving this interface to Transaction.Tests package, and inject it into fault injection `ITransactionalStateStorage`, by decorating existing `ITransactionalStateStorage`

Reply:
I was just copying the way the skewed clock injection works - if that pattern is not o.k. then the skewed clock should be moved as well.

Reply:
hmm @jason-bragg  here since he added the IClock interface. I think IClock interface is not just testing specific , it provides ability to inject other type of clock ? @jason-bragg please clarity your intention

Reply:
This should probably be implemented as an ITransactionalStateStorage decorator, but this is ok for now, we'll need to restructure in future pass.